i-VRESSE/bartender
# Add /{jobid}/directories route

Fixes #55

TODO
* [x] ~~think harder about return shape.~~ Looks a lot like [os.DirEntry](https://docs.python.org/3/library/os.html#os.DirEntry) which is ok.
* [x] add prefix, so you can list files in a sub dir only.
* [x] make async
* [x] add review instructions

To test:
1. Setup bartender with word count application
2. Follow https://github.com/i-VRESSE/bartender#word-count-example instructions
3. Try out the `GET /api/job/{jobid}/directories` route
    * It should return listing with files named: README.md, stdout.txt, stderr.txt, returncode and meta .
4. Try out the `GET /api/job/{jobid}/directories/{path}` 
   * As word count application does not have sub directories you should make one by running `mkdir /tmp/jobs/1/somedir && /tmp/jobs/1/somedir/somefile`
   * Use `somedir` as value for path parameter.
   * Should return `somefile` as file.
